astropy/coordinates/tests/test_finite_difference_velocities.py

```diff
@@ -0,0 +1,172 @@
+                                           (1*u.year, True),
+                                           (1*u.second, False),
+                                           (1*u.year, False)])
+def test_faux_lsr(dt, symmetric):
+    class LSR2(LSR):
+        obstime = TimeFrameAttribute(default=J2000)
+
+    dt = 1*u.s
+    @frame_transform_graph.transform(FunctionTransformWithFiniteDifference,
+                                     ICRS, LSR2, finite_difference_dt=dt,
+                                     symmetric_finite_difference=symmetric)
+    def icrs_to_lsr(icrs_coo, lsr_frame):
+        dt = lsr_frame.obstime - J2000
+        offset = lsr_frame.v_bary * dt.to(u.second)
+        return  lsr_frame.realize_frame(icrs_coo.data.without_differentials() + offset)
+
+    @frame_transform_graph.transform(FunctionTransformWithFiniteDifference,
+                                     LSR2, ICRS, finite_difference_dt=dt,
+                                     symmetric_finite_difference=symmetric)
+    def lsr_to_icrs(lsr_coo, icrs_frame):
+        dt = lsr_frame.obstime - J2000
+        offset = lsr_frame.v_bary * dt.to(u.second)
+        return icrs_frame.realize_frame(lsr_coo.data - offset)
+
+
+    ic = ICRS(ra=12.3*u.deg, dec=45.6*u.deg, distance=7.8*u.au,
+              pm_ra_cosdec=0*u.marcsec/u.yr, pm_dec=0*u.marcsec/u.yr,
+              radial_velocity=0*u.km/u.s)
+    lsrc = ic.transform_to(LSR2())
+
+    assert quantity_allclose(ic.cartesian.xyz, lsrc.cartesian.xyz)
+
+    idiff = ic.cartesian.differentials['s']
+    ldiff = lsrc.cartesian.differentials['s']
+    change = (ldiff.d_xyz - idiff.d_xyz).to(u.km/u.s)
+    totchange = np.sum(change**2)**0.5
+    assert quantity_allclose(totchange, np.sum(lsrc.v_bary.d_xyz**2)**0.5)
+
+
+    ic2 = ICRS(ra=120.3*u.deg, dec=45.6*u.deg, distance=7.8*u.au,
+              pm_ra_cosdec=0*u.marcsec/u.yr, pm_dec=10*u.marcsec/u.yr,
+              radial_velocity=1000*u.km/u.s)
+    lsrc2 = ic2.transform_to(LSR2())
+
+    tot = np.sum(lsrc2.cartesian.differentials['s'].d_xyz**2)**0.5
+    assert np.abs(tot.to('km/s') - 1000*u.km/u.s) < 20*u.km/u.s
+
+def test_gcrs_diffs():
+    time = Time('J2017')
+    gf = GCRS(obstime=time)
+    sung = get_sun(time)  # should have very little vhelio
+
+    # qtr-year off sun location should be the direction of ~ maximal vhelio
+    qtrsung = get_sun(time-.25*u.year)
+
+    #now we use those directions to
+    msungr = CartesianRepresentation(-sung.cartesian.xyz).represent_as(SphericalRepresentation)
+    suni = ICRS(ra=msungr.lon, dec=msungr.lat, distance=100*u.au,
+                pm_ra_cosdec=0*u.marcsec/u.yr, pm_dec=0*u.marcsec/u.yr,
+                radial_velocity=0*u.km/u.s)
+    qtrsuni = ICRS(ra=qtrsung.ra, dec=qtrsung.dec, distance=100*u.au,
+                   pm_ra_cosdec=0*u.marcsec/u.yr, pm_dec=0*u.marcsec/u.yr,
+                   radial_velocity=0*u.km/u.s)
+    sung = suni.transform_to(gf)
+    qtrsung = qtrsuni.transform_to(gf)
+
```

Comment:
It's in the direction either with or perpendicular to the Earth's orbit, but transforming to GCRS *adds in* the Earth's rotation (or *not*, if it's the perpendicular... which is the point of the test) will clarify with a comment